### PR TITLE
Feature/apiprofile header improvements

### LINKED ISSUE #1625: Organize API Profile header
I notice that we are adding new components to API Profile header and it starts to look clumsy.
### Proposition:

- [ ] Organize existing components, if needed move some components out from header
- [ ] Plan (ex. wireframe) position of future components based on backlog

### PR DESCRIPTION
**Pull request by nnn**
Closes #1625 

### Proposed changes
- Moved tour button
- Moved private/public button
- Cleaned up

### CODE REVIEW

**Comment by nnn** [comment]
Partly blocked by #1618, will be ready for a review when we decide on rating position.

**Comment by nnn** [comment]
@apinf/developers, please review.

File: apis/client/view/header/header.html [hunk 1]
```diff
@@ -2,43 +2,37 @@
...
+          {{> viewApiStatus api=api width="0.4" }}
+        </h1>
+        <ul class="list-inline view-api-meta-details">
+          <li>
+            <i class="fa fa-user"></i>
+            {{ api.getApiManagersByName }}
+          </li>
+        </ul>
+      </div>
+      <div class="page-head-starsblock pull-right">
```
Comment (brylie) [inline apis/client/view/header/header.html]: Lets rename this from 'starsblock' to something more general, since it contains more page elements than the star rating widget. E.g. page-head is fine.

File: apis/client/view/header/header.html [hunk 2]
```diff
@@ -2,43 +2,37 @@
...
+          {{> favourite api=api }}
+        {{else}}
+          <a href="#" class="star-button">
+            <div
+                class="api-card-bookmark"
+                data-toggle="tooltip"
+                data-placement="bottom"
+                title="{{_ 'catalogue_bookmarkTooltip_nonUser'}}">
+              <span class="api-card-bookmark-number">{{ api.bookmarkCount }}</span>
+              <i class="fa fa-star bookmark-default"></i>
```
Comment (brylie) [inline apis/client/view/header/header.html]: This should no longer be a star, since it is related to bookmarking. Also, the page text and button class should be changed to use the word 'bookmark' instead of 'star'. Will these changes be part of a different PR?

File: apis/client/view/visibility/visibility.js [hunk 3]
```diff
@@ -1,36 +1,35 @@
...
+    const instance = this;
+
+    // Get API Backend from data context
+    const api = this.api;
+
+    // Get ID of current service
+    const apiId = api._id;
+
+    // Check visibility status
+    const status = Apis.findOne(apiId).isPublic;
```
Comment (brylie) [inline apis/client/view/visibility/visibility.js]: This can be shortened, since the API is already part of the template data context. In order to get the template data context in a way that is reactive, use `Template.currentData()`, rather than `instance.data`.

http://blazejs.org/api/templates.html#Template-currentData

File: apis/client/view/header/header.html [hunk 4]
```diff
@@ -2,43 +2,37 @@
...
-                  data-placement="bottom"
-                  title="{{_ 'catalogue_bookmarkTooltip_nonUser'}}">
-                <span class="api-card-bookmark-number">{{ api.bookmarkCount }}</span>
-                <i class="fa fa-star bookmark-default"></i>
-              </div>
-            </a>
-          {{/if}}
-        </div>
-        {{#if api.currentUserCanEdit }}
-          {{> visibilitySwitch api=api }}
```
Comment (brylie) [inline apis/client/view/header/header.html]: I think this reduces the usablility for the *visibility toggle* feature for several reasons:
- We have lost the explicit text saying whether the API is *public* or *private*
- The *visibility toggle* has been moved to the settings page, and nested among other fields
- The *visibility toggle* button on the settings page does not say what the current state is
  - it rather says the *opposite* of what the current stat is, as a verb

**Comment by jykae** [comment]
Reviewing

**Comment by brylie** [comment]
I think we need to review this PR as a group. Ping @bajiat @Nazarah

File: apis/client/view/header/header.html (hunk 2 above)
Comment (nnn) [inline apis/client/view/header/header.html]: Yes, different PR.

**Comment by brylie** [comment]
I will defer to @jykae as to whether this gets merged.

However, I would like to have more clear discussion about changes like this, including defining related task more specifically in the future.

**Comment by nnn** [comment]
@jykae, please review.

File: apis/client/view/header/header.html (hunk 4 above)
Comment (jykae) [inline apis/client/view/header/header.html]: @brylie I agree, though visibility feature is going to have third option "partly private".
@Nazarah has wireframe for new UI.

**Comment by jykae** [comment]
@nnn This does not look too good in my eye, could API key component be on left under API logo, name, owner information? @Nazarah opinion?

nayttokuva 2016-10-03 kello 15 40 02

**Comment by brylie** [comment]
@jykae is this task still under review, or are you waiting for follow-up from @nnn?

I removed the 'ready for review', so that conflicts could be resolved.

**Comment by jykae** [comment]
Changes discussed and approved with @nnn @brylie
